src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -354,6 +354,29 @@ public int read(byte b[], int off, int len) throws IOException {
         return (capacity == nread) ? buf : Arrays.copyOf(buf, nread);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public long transferTo(OutputStream out) throws IOException {
+        if (out instanceof FileOutputStream fos) {
+            FileChannel fci = getChannel();
+            long pos = fci.position();
+            long count = fci.size() - pos;
+            if (pos >= 0 && count >= 0) {
+                FileChannel fco = fos.getChannel();
+
+                // Return if transferTo() copies enough bytes, but if it does
+                // not, then use the superclass method to copy those remaining
+                long transferred = fci.transferTo(pos, count, fco);
+                fci.position(pos + transferred);
+                if (transferred == count) {
+                    return count;
+                }
+            }
+        }
+        return super.transferTo(out);
```

Comment:
I think there is also another bug here for the case that transferTo does a partial transfer, it should be:
  return transferred + super.transferTo(out));